zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupCreateProcessor.java

```diff
@@ -53,25 +53,27 @@ public GroupCreateProcessor(
 
   @Override
   public void processNewCommand(final TypedRecord<GroupRecord> command) {
+    final var record = command.getValue();
+    final var groupId = record.getGroupId();
+    final var persistedGroup = groupState.get(groupId);
+    if (persistedGroup.isPresent()) {
+      final var errorMessage = GROUP_ALREADY_EXISTS_ERROR_MESSAGE.formatted(groupId);
+      rejectionWriter.appendRejection(command, RejectionType.ALREADY_EXISTS, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.ALREADY_EXISTS, errorMessage);
+      return;
+    }
```

Comment:
❓ Why did you choose to put this validation above the authorization check? This way we expose our state (that a group exists) to a user that might not be authorized to know about this.